src/components/ImageListContainer.js

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import {addImages, endDowloadingImages, startDowloadingImages} from "../actions";
+import {connect} from "react-redux";
+import ImageList from "./ImageList";
+import PropTypes from "prop-types";
+import {IMAGE_HEIGHT} from "../constants";
+
+class ImageListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isFetching: false,
+        };
+    }
+
+    handleScroll = (e) => {
+        const bottom = e.target.scrollingElement.scrollHeight - IMAGE_HEIGHT - e.target.scrollingElement.scrollTop <= e.target.scrollingElement.clientHeight + 1;
+        if (bottom&&!this.props.isFetching) {
```

Comment:
This way one should remember to check for `isFetching` before using `addImagesToStore` in the future. Let's move the `!isFetching` logic into the `addCat` action. For this you may use the 2nd argument of the thunk function, like this:
```js
return (dispatch, getState) => {
```